src/vs/platform/windows/electron-main/windowsMainService.ts

```diff
@@ -1498,27 +1501,31 @@ export class WindowsMainService extends Disposable implements IWindowsMainServic
 			}
 		}
 
+		if (this.userDataProfilesMainService.isEnabled()) {
+			const workspace = configuration.workspace ?? toWorkspaceIdentifier(configuration.backupPath, false);
+			const profile = await this.resolveProfileForBrowserWindow(options, workspace);
```

Comment:
Will this return sync without work in the normal case? Asking for startup perf here as its on the critical path.

Reply:
Yes, it can. But there was no change in perf. I will change it anyway.